File: javascript/beer-song/beer-song.js
const bottles_on_wall =  n=>{
  switch(n){
    case(0) : return 'No more bottles'; 
    case(1) : return '1 bottle';
    default :  return `${n} bottles`;
  }
};

const first_sentence = n => `${bottles_on_wall(n)} of beer on the wall, ${bottles_on_wall(n).toLowerCase()} of beer.`;

const second_sentence= n => {
  switch(n){
    case(0) : return 'Go to the store and buy some more, 99 bottles of beer on the wall.'; 
    case(1) : return 'Take it down and pass it around, no more bottles of beer on the wall.';
    case(2) : return 'Take one down and pass it around, 1 bottle of beer on the wall.';
    default :  return `Take one down and pass it around, ${n-1} bottles of beer on the wall.`
  }
};


export class BeerSong {
  static verse(n) {
    return `${first_sentence(n)}\n${second_sentence(n)}\n`;
  }

  static sing(begin=99, end=0) {
    let res = '';
    for(let i=end; i<= begin; ++i){
      res = `${BeerSong.verse(i)}\n${res}`;
    }
    return res.slice(0,-1);
  }
}

File: lua/beer-song/beer-song.lua


function verse(num)
	if num ==0 then
		return 'No more bottles of beer on the wall, no more bottles of beer.\nGo to the store and buy some more, 99 bottles of beer on the wall.\n'

	elseif num ==1 then
		return '1 bottle of beer on the wall, 1 bottle of beer.\nTake it down and pass it around, no more bottles of beer on the wall.\n'
	elseif num ==2 then
		return '2 bottles of beer on the wall, 2 bottles of beer.\nTake one down and pass it around, 1 bottle of beer on the wall.\n'
	else
		return tostring(num) .. ' bottles of beer on the wall, ' .. tostring(num) .. ' bottles of beer.\nTake one down and pass it around, ' .. tostring(num-1).. ' bottles of beer on the wall.\n'
	end
end



function sing(num, stop)
	stop = stop or 0
	local song =''
	for i = num, stop, -1 do 
		song = song .. verse(i) .. '\n'
	end
	song = string.sub(song, 1, -2)
	return song
end
return { verse = verse,
	sing = sing}
